ocaml/xapi/repository.ml

```diff
@@ -24,6 +24,8 @@ module UpdateIdSet = Set.Make (String)
 
 let capacity_in_parallel = 16
 let reposync_mutex = Mutex.create ()
+(* The cache below is protected by pool's current_operations locking mechanism *)
+let updates_in_cache = ref None
```

Comment:
If this is a mutable associative list then why not use a Hashtbl instead? It should be more efficient and have a more natural API to make updates to it than an associative list. If you need a functional datastructure then a reference to a Map might be better suited than an associative list (XAPI tends to use lists a lot, even when something else would be both more efficient and have a better API)

Reply:
Thanks. I changed it to `Map`.